entity-service/src/main/kotlin/com/egm/stellio/entity/repository/QueryUtils.kt

```diff
@@ -191,4 +186,79 @@ object QueryUtils {
                }
             """.trimIndent()
         }
+
+    fun prepareQueryForAuthorizedEntitiesWithAuthentication(
+        queryParams: QueryParams,
+        offset: Int,
+        limit: Int
+    ): String {
+        val (_, expandedType, _, q, _) = queryParams
+        val matchEntityClause = buildMatchEntityClause(expandedType, prefix = "")
+        val authTerm = buildAuthTerm(q)
+        val matchAuthorizedEntitiesClause =
+            """
+            CALL {
+                MATCH (user)
+                WHERE user.id IN ${'$'}userAndGroupIds
+                WITH user
```

Comment:
not sure you still need this line (was needed before because of the `CALL` after)